File: Datos/futbol.php
<?php
// Inicialización de variables
echo "Ingrese el nombre del equipo visitante: ";
$equipoVisitanteNombre = trim(fgets(STDIN));
echo "Ingrese el nombre del equipo local: ";
$equipoLocalNombre = trim(fgets(STDIN));

$golesEquipoVisitante = 0;
$golesEquipoLocal = 0;

// Función para mostrar el marcador
function mostrarMarcador($equipoVisitanteNombre, $golesEquipoVisitante, $equipoLocalNombre, $golesEquipoLocal) {
    echo "Marcador: $equipoVisitanteNombre $golesEquipoVisitante vs $equipoLocalNombre $golesEquipoLocal\n";
}

// Función para actualizar el marcador
function actualizarMarcador(&$golesEquipoVisitante, &$golesEquipoLocal, $equipoVisitanteNombre, $equipoLocalNombre) {
    echo "¿Qué equipo anotó un gol? (visitante/local): ";
    $equipo = trim(fgets(STDIN));
    if (strtolower($equipo) == strtolower($equipoVisitanteNombre)) {
        $golesEquipoVisitante++;
    } elseif (strtolower($equipo) == strtolower($equipoLocalNombre)) {
        $golesEquipoLocal++;
    } else {
        echo "Entrada no válida. Por favor, ingrese 'visitante' o 'local'.\n";
    }
}

// Mostrar el marcador inicial
mostrarMarcador($equipoVisitanteNombre, $golesEquipoVisitante, $equipoLocalNombre, $golesEquipoLocal);

// Actualizar el marcador en un bucle
while (true) {
    actualizarMarcador($golesEquipoVisitante, $golesEquipoLocal, $equipoVisitanteNombre, $equipoLocalNombre);
    mostrarMarcador($equipoVisitanteNombre, $golesEquipoVisitante, $equipoLocalNombre, $golesEquipoLocal);
    echo "¿Desea continuar actualizando el marcador? (si/no): ";
    $continuar = trim(fgets(STDIN));
    if (strtolower($continuar) != "si") {
        break;
    }
}

echo "Marcador final:\n";
mostrarMarcador($equipoVisitanteNombre, $golesEquipoVisitante, $equipoLocalNombre, $golesEquipoLocal);
?>

File: Datos/futbol.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strings"
)

func main() {
    reader := bufio.NewReader(os.Stdin)

    // Inicialización de variables
    fmt.Print("Ingrese el nombre del equipo visitante: ")
    equipoVisitanteNombre, _ := reader.ReadString('\n')
    equipoVisitanteNombre = strings.TrimSpace(equipoVisitanteNombre)

    fmt.Print("Ingrese el nombre del equipo local: ")
    equipoLocalNombre, _ := reader.ReadString('\n')
    equipoLocalNombre = strings.TrimSpace(equipoLocalNombre)

    golesEquipoVisitante := 0
    golesEquipoLocal := 0

    // Función para mostrar el marcador
    mostrarMarcador := func() {
        fmt.Printf("Marcador: %s %d vs %s %d\n", equipoVisitanteNombre, golesEquipoVisitante, equipoLocalNombre, golesEquipoLocal)
    }

    // Función para actualizar el marcador
    actualizarMarcador := func() {
        fmt.Print("¿Qué equipo anotó un gol? (visitante/local): ")
        equipo, _ := reader.ReadString('\n')
        equipo = strings.TrimSpace(strings.ToLower(equipo))
        if equipo == strings.ToLower(equipoVisitanteNombre) {
            golesEquipoVisitante++
        } else if equipo == strings.ToLower(equipoLocalNombre) {
            golesEquipoLocal++
        } else {
            fmt.Println("Entrada no válida. Por favor, ingrese 'visitante' o 'local'.")
        }
    }

    // Mostrar el marcador inicial
    mostrarMarcador()

    // Actualizar el marcador en un bucle
    for {
        actualizarMarcador()
        mostrarMarcador()
        fmt.Print("¿Desea continuar actualizando el marcador? (si/no): ")
        continuar, _ := reader.ReadString('\n')
        continuar = strings.TrimSpace(strings.ToLower(continuar))
        if continuar != "si" {
            break
        }
    }

    fmt.Println("Marcador final:")
    mostrarMarcador()
}
